src/NHibernate.Test/TypesTest/StringTypeWithLengthFixture.cs

```diff
@@ -145,164 +204,92 @@ private void AssertFailedInsertExceptionDetailsAndEmptyTable(Exception ex)
 			}
 		}
 
-
 		[Test]
 		public void CriteriaLikeParameterCanExceedColumnSize()
 		{
-			Exception exception = CatchException<Exception>(
-				() =>
-				{
-					using (ISession s = OpenSession())
-					using (s.BeginTransaction())
-					{
-						s.Save(new StringClass {Id = 1, StringValue = "AAAAAAAAAB"});
-						s.Save(new StringClass {Id = 2, StringValue = "BAAAAAAAAA"});
-
-						var aaItems =
-							s.CreateCriteria<StringClass>()
-							 .Add(Restrictions.Like("StringValue", "%AAAAAAAAA%"))
-							 .List();
-
-						Assert.That(aaItems.Count, Is.EqualTo(2));
-					}
-				});
+			using (ISession s = OpenSession())
+			using (s.BeginTransaction())
+			{
+				s.Save(new StringClass {Id = 1, StringValue = "AAAAAAAAAB"});
+				s.Save(new StringClass {Id = 2, StringValue = "BAAAAAAAAA"});
 
+				var aaItems =
+					s.CreateCriteria<StringClass>()
+					 .Add(Restrictions.Like("StringValue", "%AAAAAAAAA%"))
+					 .List();
 
-			// This test fails against the ODBC driver. The driver would need to be override to allow longer parameter sizes than the column.
-			AssertExpectedFailureOrNoException(
-				exception,
-				(Sfi.ConnectionProvider.Driver is OdbcDriver));
```

Comment:
Was no more failing since 131b97e9971608e5f90b31cd57e09b8d7935c4a8 which was losing the `Size` set on parameter. But since "expand parameter" is now fixed, it would fail again if we do not do NH-4083.